src/components/OrderSuccesful/OrderSuccesful.jsx

```diff
@@ -1,24 +1,100 @@
-import { useSelector } from 'react-redux';
-import { BsFillCheckCircleFill } from 'react-icons/bs'
-import './OrderSuccesful.css'
-import languageSelector from '../../assets/languages/languageSelector';
+import { useSelector, useDispatch } from "react-redux";
+import { BsFillCheckCircleFill } from "react-icons/bs";
+import "./OrderSuccesful.css";
+import languageSelector from "../../assets/languages/languageSelector";
+import { useEffect, useState } from "react";
+import { API_URL } from "../../constants/apiUrl";
+import axios from "axios";
+import Cookies from "universal-cookie";
+import { RESET_ORDER } from "../../store/reducers/Order.reducer";
+import { type } from "@testing-library/user-event/dist/type";
+import {
+  RESET_CART,
+  UPLOADSUBTOTAL,
+} from "../../store/reducers/CartContent.reducer";
+import { useNavigate } from "react-router-dom";
+import { toast, ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
 
-const OrderSuccesful = () =>{
-  const language = useSelector(state=> state.languageReducer);
-  const orderID = useSelector(state => state.orderReducer.orderID);
+const OrderSuccesful = () => {
+  const language = useSelector((state) => state.languageReducer);
+  const orderData = useSelector((state) => state.orderReducer);
+  const price = useSelector((state) => state.subtotalReducer);
+  const cart = useSelector((state) => state.cartReducer);
+  const cookies = new Cookies();
+  const [orderID, setOrderID] = useState("");
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
 
-  return(
-    <section className='orderSuccesful__container'>
+  useEffect(() => {
+    const dishList = [];
+
+    axios
+      .post(
+        `${API_URL}/orders`,
+        {
+          price,
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${cookies.get("token")}`,
+          },
+        }
+      )
+      .then((res) => {
+        setOrderID(res.data.data.id_order);
+        for (const dish of cart) {
+          dishList.push({
+            quantity: dish.quantity,
+            ordersId_order: res.data.data.id_order,
+            dishesId_dish: dish.id,
+            restaurantsId_restaurant: dish.id_restaurant,
+          });
+        }
+        axios
+          .post(`${API_URL}/orders-details/several`, {
+            dishList,
+          })
+          .then(() => {
+            toast.success("Going back to the main page", {
```

Comment:
I consider that toast messages should be better in constants for importing when are necessaries.